### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by electro199**
ffmpeg path , none type error , fixed and removed unusable settings

### CODE REVIEW

File: TTS/engine_wrapper.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from utils.console import print_step, print_substep
 from utils.voice import sanitize_text
 
-DEFAULT_MAX_LENGTH: int = 50  # video length variable
+DEFAULT_MAX_LENGTH: int = 29  # video length variable
```
Comment (OpenSourceSimon) [inline TTS/engine_wrapper.py:20]: Why?
```suggestion
DEFAULT_MAX_LENGTH: int = 50  # Video length variable, edit this on your own risk. It should work, but it's not supported
```

File: examples/final_video.mp4
Comment (OpenSourceSimon) [inline examples/final_video.mp4:1]: Can you remove this? It's not used

File: run.bat
Comment (OpenSourceSimon) [inline run.bat:1]: This won't install the dependencies and Python needs to be installed already, so what's the benefit?

File: utils/ffmpeg_install.py [hunk 2]
```diff
@@ -60,7 +58,8 @@ def ffmpeg_install():
     try:
         # Try to run the FFmpeg command
         subprocess.run(['ffmpeg', '-version'], check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        print('FFmpeg is installed on this system! If you are seeing this error for the second time, restart your computer.')
+        if not os.path.exists("./results") :
```
Comment (OpenSourceSimon) [inline utils/ffmpeg_install.py:61]: Why this if statement?

File: video_creation/final_video.py [hunk 3]
```diff
@@ -337,7 +337,7 @@ def make_final_video(
         text=text,
         x=f"(w-text_w)",
         y=f"(h-text_h)",
-        fontsize=12,
+        fontsize=5,
```
Comment (OpenSourceSimon) [inline video_creation/final_video.py:340]: Great!

File: video_creation/screenshot_downloader.py [hunk 4]
```diff
@@ -210,12 +210,12 @@ def get_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: int):
                 if idx >= screenshot_num:
                     break
 
-                if page.locator('[data-testid="content-gate"]').is_visible():
-                    page.locator('[data-testid="content-gate"] button').click()
+                    if page.locator('[data-testid="content-gate"]').is_visible():
+                        page.locator('[data-testid="content-gate"] button').click()
 
-                page.goto(f'https://reddit.com{comment["comment_url"]}', timeout=0)
+                    page.goto(f'https://reddit.com{comment["comment_url"]}', timeout=0)
```
Comment (OpenSourceSimon) [inline video_creation/screenshot_downloader.py:216]: Nope this will break a lot of stuff, why?

File: video_creation/screenshot_downloader.py [hunk 5]
```diff
@@ -240,3 +240,28 @@ def get_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: int):
...
+        txtcolor = (0, 0, 0)
+    if settings.config["settings"]["storymode"]:
+        if settings.config["settings"]["storymodemethod"] == 1:
+            # for idx,item in enumerate(reddit_object["thread_post"]):
+            imagemaker(theme=bgcolor, reddit_obj=reddit_object, txtclr=txtcolor)
+
+    if (
+        settings.config["settings"]["storymodemethod"] == 0
+        or not settings.config["settings"]["storymode"]
+    ):
```
Comment (OpenSourceSimon) [inline video_creation/screenshot_downloader.py:266]: What?! Why here?

**Review by OpenSourceSimon — CHANGES_REQUESTED** [review]
Thanks for the PR, but there are definitely some things that needs to be fixed

File: utils/ffmpeg_install.py (hunk 2 above)
Comment (electro199) [reply utils/ffmpeg_install.py:65]: i will take a look

File: video_creation/screenshot_downloader.py (hunk 5 above)
Comment (electro199) [reply video_creation/screenshot_downloader.py:266]: reverting

File: utils/ffmpeg_install.py (hunk 2 above)
Comment (electro199) [reply utils/ffmpeg_install.py:65]: to check if person is newly installing the bot

File: examples/final_video.mp4
Comment (OpenSourceSimon) [inline examples/final_video.mp4:1]: Can you remove this? It's not used

File: run.bat
Comment (electro199) [reply run.bat:1]: it is to run the bot without opening the cmd just click

File: main.py [hunk 6]
```diff
@@ -75,31 +72,32 @@ def run_many(times) -> None:
...
-def shutdown():
-    try:
-        redditid
-    except NameError:
-        print("Exiting...")
-        exit()
-    else:
+def shutdown() -> NoReturn:
+
+    if "redditid" in globals() :
```
Comment (OpenSourceSimon) [inline main.py:77]:
```suggestion
    if "redditid" in globals():
```

**Review by OpenSourceSimon — APPROVED** [review]

File: run.bat [hunk 7]
```diff
@@ -6,15 +6,13 @@ if exist "%VENV_DIR%" (
     call "%VENV_DIR%\Scripts\activate.bat"
 )
 
+echo Installing dependencies...
+pip install -r requirements.txt
+
```
Comment (electro199) [inline run.bat:11]: don't add this

Comment (electro199) [reply run.bat:11]: it is to run not to setup

File: utils/ffmpeg_install.py [hunk 8]
```diff
@@ -77,10 +75,13 @@ def ffmpeg_install():
         # Try to run the FFmpeg command
         subprocess.run(['ffmpeg', '-version'], check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     except FileNotFoundError as e:
-        if not os.path.exists("./results") :
+        # Check if there's ffmpeg.exe in the current directory
+        if os.path.exists("./ffmpeg.exe"):
```
Comment (electro199) [inline utils/ffmpeg_install.py:79]: not making any sense

File: utils/posttextparser.py [hunk 9]
```diff
@@ -9,19 +10,19 @@
...
+    text: str = re.sub("\n", " ", obj)
     try:
         nlp = spacy.load("en_core_web_sm")
     except OSError as e:
         if not tried:
             os.system("python -m spacy download en_core_web_sm")
-            return posttextparser(obj,tried=True)
-        
-        print_step(" The spacy model can't load. You need to install it with the command \npython -m spacy download en_core_web_sm ")
+            time.sleep(5)
```
Comment (electro199) [inline utils/posttextparser.py:20]: why sleep ?

File: run.bat (hunk 7 above)
Comment (OpenSourceSimon) [reply run.bat:11]:
```suggestion
```

File: utils/ffmpeg_install.py (hunk 8 above)
Comment (OpenSourceSimon) [reply utils/ffmpeg_install.py:79]: Yes it is, to check if it already has been downloaded

File: utils/posttextparser.py (hunk 9 above)
Comment (OpenSourceSimon) [reply utils/posttextparser.py:20]: Since the command needs some time to execute

File: main.py [hunk 10]
```diff
@@ -73,30 +71,24 @@ def run_many(times) -> None:
...
     directory = Path().absolute()
     config = settings.check_toml(
         f"{directory}/utils/.config.template.toml", f"{directory}/config.toml"
     )
-    if not config :
-        sys.exit()
+    config is False and sys.exit()
         
     if (
         not settings.config["settings"]["tts"]["tiktok_sessionid"]
```
Comment (electro199) [inline main.py:94]: where is my change ?

File: run.bat (hunk 7 above)
Comment (OpenSourceSimon) [reply run.bat:11]: Sure

File: run.bat [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+set VENV_DIR=.venv
+
+if exist "%VENV_DIR%" (
+    echo Activating virtual environment...
+    call "%VENV_DIR%\Scripts\activate.bat"
+)
+
+echo Installing dependencies...
+pip install -r requirements.txt
+
```
Comment (OpenSourceSimon) [inline run.bat:11]:
```suggestion
```

**Review by OpenSourceSimon — APPROVED** [review]

File: main.py (hunk 10 above)
Comment (OpenSourceSimon) [reply main.py:94]: This line is efficient enough, I don't know why you changed it

Comment (electro199) [reply main.py:94]: @OpenSourceSimon

File: utils/posttextparser.py (hunk 9 above)
Comment (electro199) [reply utils/posttextparser.py:20]: it is blocking code not async

**Pull request merged**